notification-service/src/main/java/org/arpha/notificationservice/service/sender/EmailNotificationSender.java

```diff
@@ -0,0 +1,49 @@
+package org.arpha.notificationservice.service.sender;
+
+import com.azure.communication.email.EmailClient;
+import com.azure.communication.email.models.EmailAddress;
+import com.azure.communication.email.models.EmailMessage;
+import com.azure.communication.email.models.EmailSendResult;
+import com.azure.core.util.polling.SyncPoller;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.arpha.domain.type.notification.configuration.EmailNotificationProperties;
+import org.arpha.notificationservice.configuration.properties.AzureCommunicationProperties;
+import org.arpha.notificationservice.mapper.NotificationMapper;
+import org.arpha.notificationservice.repository.NotificationRepository;
+import org.arpha.utils.Boxed;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class EmailNotificationSender implements NotificationSender<EmailNotificationProperties> {
+
+    private final NotificationMapper notificationMapper;
+    private final NotificationRepository notificationRepository;
+    private final EmailClient emailClient;
+    private final AzureCommunicationProperties azureCommunicationProperties;
+
+    @Async
+    @Override
+    public void sendNotification(EmailNotificationProperties notificationProperties) {
+        Boxed.of(notificationProperties)
+                .to(this::sendMessage)
+                .to(emailResult -> notificationMapper.mapToNotification(notificationProperties, emailResult))
+                .apply(notificationRepository::save);
+    }
+
+    private EmailSendResult sendMessage(EmailNotificationProperties notificationProperties) {
+        EmailMessage emailMessage = new EmailMessage()
+                .setSenderAddress(azureCommunicationProperties.senderEmail())
+                .setToRecipients(new EmailAddress(notificationProperties.getRecipientEmail()))
+                .setSubject(notificationProperties.getSubject())
+                .setBodyHtml(notificationProperties.getBody());
```

Comment:
mapper?

Reply:
I am not a big fan of using mappers in such cases. I dont want to turn mappers into some kind of allmighty entities.
So, whenever I want to call third-party API with some request, assembling of which requires additional dependencies (AzureCommunicationProperties in this particular case), I would rather assemble it directly in sendMessage method instead of moving it to mapper

Reply:
Also, in my opinion complex mappers, like those where we use qualifiedByName, do not make much sense.
Why? Well, it just increases code complexity and does not affect code cleanliness (IMHO). Code should be intuitive and easy to read